### PR TITLE
Make `data.create_image_fetcher` private

### LINKED ISSUE #6854: `data.create_image_fetcher` should be private
### Description:

https://github.com/scikit-image/scikit-image/blob/c90bfccbc930262eebd1149b10660bfa4f56ec61/skimage/data/__init__.pyi#L16

has no docstring and is probably intended as a private initializer for pooch. So remove it from the PYI file and probably add an underscore to the function name to be totally explicit about it. :wink: 

### Way to reproduce:

```python
import skimage as ski
assert ski.data.create_image_fetcher.__doc__   # passes
```

### Version information:

```Shell
3.10.10 (main, Mar  5 2023, 22:26:53) [GCC 12.2.1 20230201]
Linux-6.2.7-arch1-1-x86_64-with-glibc2.37
scikit-image version: 0.21.0rc0.dev0+git20230323.c90bfccbc
numpy version: 1.24.1
```

### PR DESCRIPTION
**Pull request by seanpquinn**
## Description
Added `create_image_fetcher` to module privates, closes #6854

## Test plan

```bash
$ meson build --prefix=.\build;meson install -C build --destdir out/;ninja -C build
$ pytest.exe --pyargs build/out/Lib/site-packages/skimage/
```

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
  `doc/release/release_dev.rst`.
- There is a bot to help automate backporting a PR to an older branch. For
  example, to backport to v0.19.x after merging, add the following in a PR
  comment: `@meeseeksdev backport to v0.19.x`
- To run benchmarks on a PR, add the `run-benchmark` label. To rerun, the label
  can be removed and then added again. The benchmark output can be checked in
  the "Actions" tab.

### CODE REVIEW

**Review by mkcor — APPROVED** [review]
Thank you for this contribution, @seanpquinn! Would you be able to resolve the current conflicts in `skimage/data/_fetchers.py`?

**Comment by seanpquinn** [comment]
> Thank you for this contribution, @seanpquinn! Would you be able to resolve the current conflicts in `skimage/data/_fetchers.py`?

Oh, sure. Hmm, this PR is pretty stale, and the change was fairly benign.

I'm also a little confused now: @mkcor you seem to have commandeered things completely with #6950 ?

**Comment by seanpquinn** [comment]
At any rate, rebased my original PR.

**Comment by stefanv** [comment]
Sorry, @seanpquinn, looks like @mkcor struggled to push to your branch. Thanks for updating!

**Comment by mkcor** [comment]
> I'm also a little confused now: @mkcor you seem to have commandeered things completely with #6950 ? What is your preference for going forward? Shall I resolve, or is your PR going to be used?
> 
> Strangely enough, this would be the 2nd time this PR is duplicated...

Yes, sorry for the untimely changes! Since this PR addresses #6854, I thought you would receive notifications (well, I know it depends on your settings) showing https://github.com/scikit-image/scikit-image/issues/6854#issuecomment-1557835574 and what follows. I started another PR just so the changes could ship with the new release (v0.21).

I wanted to push to your existing branch, but the GitHub interface said I had to use the command line, which I did, I pulled your branch locally, but since it's "main", it turned out to be pretty messy -- please create feature branches with descriptive names! :pray:

File: skimage/data/_fetchers.py [hunk 1]
```diff
@@ -124,8 +124,12 @@ def create_image_fetcher():
     return image_fetcher, data_dir
 
 
-image_fetcher, data_dir = create_image_fetcher()
+_image_fetcher, data_dir = _create_image_fetcher()
 
+if _image_fetcher is None:
```
[the remark this answers is not in the document]
Comment (mkcor) [reply skimage/data/_fetchers.py:129]: These should be gone. The merge isn't trivial -- I spent a little while yesterday to come up with 0408c1d589d05df136b9abc082827658e8d08855, i.e., reconciling @seanpquinn's changes with what @lagru did in #6886. So, if we end up merging this PR, we should ensure the diffs match those at #6950.

[the remark this answers is not in the document]
Comment (seanpquinn) [reply skimage/data/_fetchers.py]: Hopefully resolved, may be worth double check

**Comment by stefanv** [comment]
Thank you for your contribution, @seanpquinn!

**Pull request merged**